test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -564,40 +504,168 @@ static int prepare_input_packet(const alg_test_param_t *param,
 	return 0;
 }
 
-static void check_output_packet_data(odp_packet_t pkt,
-				     const alg_test_param_t *param)
-{
-	crypto_test_reference_t *ref = param->ref;
-	uint8_t *expected_bytes;
+#define MAX_IGNORED_RANGES 3
 
-	/* crypto output is undefined if authentication fails */
-	if (param->wrong_digest)
+/*
+ * Output packet parts that we ignore since they have undefined values
+ */
+typedef struct ignore_t {
+	uint32_t byte_offset;	/* offset to a byte which has bits to be ignored */
```

Comment:
With bit mode, are we testing non-byte aligned memory also?

This may require a capability. MRVL doesn't support non-byte aligned and as per PDCP non-byte aligned is not required.

Reply:
No, we are not testing non-byte aligned ranges in bit mode. ODP API requires that the starting offset of cipher range or auth range is byte aligned even with algorithms that operate in bit mode. This what the API spec says:

```
         * Note: data buffer MUST start on the byte boundary, using offset
         * which is not divisible by 8 is unsupported and will result in
         * unspecified behaviour.
```

The length of the range in bit mode algorithms does not need to be a multiple of 8 bits. In that case the bits of the last byte that are undefined after the operation, which is that this byte_offset and byte_mask in ingore_t are trying to handle.

Reply:
> The length of the range in bit mode algorithms does not need to be a multiple of 8 bits.
MRVL does not support such operations. Can this be protected with some capability check?

Reply:
Thanks Janne for clarification. That capability flag should do. We will make adjustments if required.